### ISSUE TITLE
Use type keyword in imports/exports for generations

### ISSUE THREAD

**Issue by jogelin** [anymaniax/orval#288]
Since typescript 3.8, it is possible to specify the keyword `type` for exporting and importing a typescript type.

This allows transpiler like babel to ignore the import and make the transpilation faster.

**Comment by melloware** [comment]
I don't understand this ticket. Isn't Orval already exporting types ?

**Comment by jogelin** [comment]
> I don't understand this ticket. Isn't Orval already exporting types ?

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export

**Comment by melloware** [comment]
I still don't understand I am looking at my generated code and its doing this already. Example below...

```ts
import { useQuery, useMutation, UseQueryOptions, UseMutationOptions, QueryFunction, MutationFunction, UseQueryResult, QueryKey } from "react-query";
import type {
    LoggersName2200,
    LoggersName200,
    Links200,
    Threaddump2200,
    Scheduledtasks200,
    Metrics200,
    MetricsRequiredMetricName200,
    Mappings200,
    Loggers200,
    Liquibase200,
    Info200,
    Heapdump200,
    Health200,
    HealthPath200,
    Env200,
    EnvToMatch200,
    Configprops200,
    ConfigpropsPrefix200,
    Conditions200,
    Caches200,
    Caches2200,
    CachesCache200,
    CachesCache2200,
    Beans200,
} from "./authorization.schemas";
```

So can you PLEASE be clear on what Orval is currently NOT doing?

**Comment by jogelin** [comment]
@melloware Indeed, it seems that internally, the `type` keyword is used.

The export `type` then could be used to expose schemas.

But really not mandatory IMO, you close this issue if you think this is not adding value

**Comment by melloware** [comment]
yeah i don't mind researching it but it seems like Orval is already doing what you need so you can close this ticket.
